lib/log.js

```diff
@@ -0,0 +1,165 @@
+'use strict'
+
+const procLog = require('proc-log')
+const { format } = require('util')
+
+// helper to emit log messages with a predefined prefix
+const logLevels = Object.keys(procLog).filter((k) => typeof procLog[k] === 'function')
+const withPrefix = (prefix) => logLevels.reduce((acc, level) => {
+  acc[level] = (...args) => procLog[level](prefix, ...args)
+  return acc
+}, {})
+
+// very basic ansi color generator
+const COLORS = {
+  wrap: (str, colors) => {
+    const codes = colors.filter(c => typeof c === 'number')
+    return `\x1b[${codes.join(';')}m${str}\x1b[0m`
+  },
+  inverse: 7,
+  fg: {
+    black: 30,
+    red: 31,
+    green: 32,
+    yellow: 33,
+    blue: 34,
+    magenta: 35,
+    cyan: 36,
+    white: 37
+  },
+  bg: {
+    black: 40,
+    red: 41,
+    green: 42,
+    yellow: 43,
+    blue: 44,
+    magenta: 45,
+    cyan: 46,
+    white: 47
+  }
+}
+
+class Logger {
+  #buffer = []
```

Comment:
observation: min engine is v12 in package.json, which is when private fields were added to node.